gapis/resolve/command_tree.go

```diff
@@ -72,30 +72,28 @@ func (t *commandTree) index(indices []uint64) (api.SpanItem, api.SubCmdIdx) {
 	return group, subCmdRootID
 }
 
-func (t *commandTree) indices(idx []uint64, preferGroup bool) []uint64 {
+func (t *commandTree) indices(cmdIdx []uint64, preferGroup bool) []uint64 {
```

Comment:
This is me trying to understand the code below.

Do I understand correctly that ```cmdIdx``` is a list of indices that indicate the path to a given command and what this function is doing is actually finding the list of positions within the corresponding group (either CmdIdGroup or SubCmdRoot) for each of the indices? How do we ensure that ```cmdIdx``` always finishes in a leaf node? Also, what happens if we end the loop and we're still in some intermediate group?

Reply:
`cmdIdx` is "the command ID", but the server is in this odd state, where we treat command IDs and sub-command IDs  differently. In most places, it's not a problem, but here is where some of the brokenness surfaces. This is one of the planned re-factors.

This function's job is to figure out which node (which is also represented by a list of numbers) in the tree is best one to use for the given command/sub-command. So, yes, it's trying to find the best `CmdIdGroup`, `SubCmdRoot`, or `CmdIDRange`.

It doesn't have to finish at a leaf, e.g. command `[10 0 1]` refers to the second command buffer in the submit, this will return the node created for that command buffer. Or if the command given is the last one in a group, the group is chosen, rather than the command itself.

We actually never end the loop early, because of the built in assumptions that `SubCmdRoot` nodes only contain other `SubCmdRoot` or `CmdIdGroup` nodes and `CmdIdGroup` nodes only contain other `CmdIDGroup` or `CmdIDRange` nodes.